x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/FixedAutoscalingDeciderService.java

```diff
@@ -88,23 +101,41 @@ public static class FixedReason implements AutoscalingDeciderResult.Reason {
 
         private final ByteSizeValue storage;
         private final ByteSizeValue memory;
+        private final Float processors;
         private final int nodes;
 
-        public FixedReason(ByteSizeValue storage, ByteSizeValue memory, int nodes) {
+        public FixedReason(ByteSizeValue storage, ByteSizeValue memory, int nodes, Float processors) {
             this.storage = storage;
             this.memory = memory;
             this.nodes = nodes;
+            this.processors = processors;
+            if (processors != null && processors < 0) {
+                throw new IllegalArgumentException("[processors] must be a non-negative number");
+            }
         }
 
         public FixedReason(StreamInput in) throws IOException {
             this.storage = in.readOptionalWriteable(ByteSizeValue::new);
             this.memory = in.readOptionalWriteable(ByteSizeValue::new);
             this.nodes = in.readInt();
+            if (in.getVersion().onOrAfter(Version.V_8_4_0)) {
+                this.processors = in.readOptionalFloat();
+            } else {
+                this.processors = null;
+            }
         }
 
         @Override
         public String summary() {
-            return "fixed storage [" + storage + "] memory [" + memory + "] nodes [" + nodes + "]";
+            return String.format(
+                Locale.ROOT,
+                // We allow processors to be optional in the output for API backwards compatibility
+                "fixed storage [%s] memory [%s] nodes [%d]%s",
+                storage,
+                memory,
+                nodes,
+                Optional.ofNullable(processors).map(i -> " processors [" + i + "]").orElse("")
```

Comment:
Let us keep nodes at then end instead. There is no API compatibility guarantee on this message.

```suggestion
                "fixed storage [%s] memory [%s] processors [%s] nodes [%d]",
                storage,
                memory,
                processors,
                nodes
```

Reply:
@henningandersen this technically breaks our BWC as the reason for old API calls is now different.

Reply:
The rest tests probably shouldn't do a perfect string match on this field anyways...I will mute them for this PR and update.